sympy/tensor/tests/test_tensor.py

```diff
@@ -1478,6 +1478,23 @@ def test_valued_tensor_expressions():
     assert expr5.data.expand() == 28*E*x1 + 12*px*x1 + 20*py*x1 + 28*pz*x1 + 136*x2 + 3*x3
 
 
+def test_valued_tensor_add_scalar():
+    numpy = import_module("numpy")
+    if numpy is None:
+        skip("numpy not installed.")
+
+    (A, B, AB, BA, C, Lorentz, E, px, py, pz, LorentzD, mu0, mu1, mu2, ndm, n0, n1,
+     n2, NA, NB, NC, minkowski, ba_matrix, ndm_matrix, i0, i1, i2, i3, i4) = _get_valued_base_test_variables()
+
+    # one scalar summand after the contracted tensor
+    expr1 = A(i0)*A(-i0) - (E**2 - px**2 - py**2 - pz**2)
+    assert expr1.data == 0
+
+    # multiple scalar summands in front of the contracted tensor
+    expr2 = E**2 - px**2 - py**2 - pz**2 - A(i0)*A(-i0)
+    assert expr2.data == 0
+
+
```

Comment:
@Upabjojr, I have only included two tests:
- one scalar summand after the contracted tensor
- multiple scalar summands in front of the contracted tensor

Another test with multiple scalar summands after the contracted tensor fails because of issue #11525.